CONTRIBUTING.md

```diff
@@ -13,13 +13,130 @@ resources to make it easier to get your contribution accepted.
+$ user={your github profile name}
+```
+
+Create your clone:
+
+```sh
+$ mkdir -p $working_dir
+$ cd $working_dir
+$ git clone https://github.com/$user/tikv.git
+$ cd $working_dir/tikv
+```
+
+View the remote info:
+
+```sh
+$ git remote -v
+origin	https://github.com/$user/tikv.git (fetch)
+origin	https://github.com/$user/tikv.git (push)
+
+$ git remote add upstream https://github.com/pingcap/tikv.git
+$ git remote -v
+origin	https://github.com/$user/tikv.git (fetch)
+origin	https://github.com/$user/tikv.git (push)
+upstream	https://github.com/pingcap/tikv.git (fetch)
+upstream	https://github.com/pingcap/tikv.git (push)
+
+# Never push to upstream master since you do not have write access.
+$ git remote set-url --push upstream no_push
+$ git remote -v
+origin	https://github.com/$user/tikv.git (fetch)
+origin	https://github.com/$user/tikv.git (push)
+upstream	https://github.com/pingcap/tikv.git (fetch)
+upstream	no_push (push)
+```
+
+### Step 3: Branch
+
+Get your local master up to date:
+
+```sh
+cd $working_dir/tikv
+git fetch upstream
+git checkout master
+git rebase upstream/master
+```
+
+Branch from master:
+
+```sh
+git checkout -b myfeature
+```
+
+### Step 4: Develop
+
+#### Edit the code
+
+You can now edit the code on the `myfeature` branch.
+
+#### Run stand-alone mode
+
+```sh
+$ make build
+```
+
+#### Run Test
+
+When you're ready to test out your changes, use the `dev` task. It will format your codebase, build with clippy enabled, and run tests. This should run without failure before you create a PR.
+
+```sh
+$ make dev
+```
+
 - Run tests and make sure all the tests are passed.
-- Make sure your commit messages are in the proper format (see below).
-- Push your changes to a topic branch in your fork of the repository.
-- Submit a pull request.
-- Your PR must receive LGTMs from two maintainers.
+- Make sure your commit messages are in the proper format.
+
+### Step 5: Keep your branch in sync
+
+```sh
+# While on your myfeature branch.
+cd $working_dir/tikv
+git fetch upstream
+git rebase upstream/master
+```
+
+### Step 6: Commit
+
+Commit your changes.
+
+```sh
+git commit
+```
+
+### Step 7: Push
+
+When ready to review (or just to establish an offsite backup or your work), push your branch to your fork on github.com:
+
+```sh
+git push -f origin myfeature
+```
+
+### Step 8: Create a pull request
+
+1. Visit your fork at [https://github.com/$user/tikv](https://github.com/$user/tikv) (replace $user obviously).
+2. Click the Compare & pull request button next to your `myfeature` branch.
+
+### Step 9: Get a code review
+
+Once your pull request has been opened, it will be assigned to at least two reviewers. Those reviewers will do a thorough code review, looking for correctness, bugs, opportunities for improvement, documentation and comments, and style.
+
+Commit changes made in response to review comments to the same branch on your fork.
+
+Very small PRs are easy to review. Very large PRs are very difficult to review.
```

Comment:
"If you notice your feature is fixing multiple things, or the feature can be broken now, please try to split it into multiple PRs"